.github/workflows/main.yaml

```diff
@@ -43,33 +85,167 @@ jobs:
+        env:
+          RANCHER_HOSTNAME: ${{ needs.installation.outputs.PUBLIC_IP}}.sslip.io
+          CATTLE_TEST_CONFIG: ${{ github.workspace }}/cattle-config-provisioning.yaml
         run: |
-          echo "Hello from the runner; hostname $(hostname -A)"
+          make e2e-provisioning-tests
+            
+  importing-tests:
+    if: ${{ github.event.inputs.run_importing_tests == 'true' }}
+    needs: installation
+    runs-on: self-hosted
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+            
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: go.mod
+
+      - name: Authenticate to GCP
+        uses: 'google-github-actions/auth@v1'
+        with:
+          credentials_json: '${{ env.GCP_CREDENTIALS }}'
+                
+      - name: Setup gcloud
+        uses: google-github-actions/setup-gcloud@v1
+
+      - name: Importing cluster tests
+        env:
+          RANCHER_HOSTNAME: ${{ needs.installation.outputs.PUBLIC_IP}}.sslip.io
+          CATTLE_TEST_CONFIG: ${{ github.workspace }}/cattle-config-import.yaml
+        run: |
+          make e2e-import-tests
+            
+  support-matrix-provisioning-tests:
+    if: ${{ github.event.inputs.run_support_matrix_provisioning_tests == 'true' }}
+    needs: installation
+    runs-on: self-hosted
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+            
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: go.mod
+            
+      - name: Support matrix tests
+        env:
+          RANCHER_HOSTNAME: ${{ needs.installation.outputs.PUBLIC_IP}}.sslip.io
+          CATTLE_TEST_CONFIG: ${{ github.workspace }}/cattle-config-provisioning.yaml
+        run: |
+          make e2e-support-matrix-tests
+
+  support-matrix-importing-tests:
+    if: ${{ github.event.inputs.run_support_matrix_importing_tests == 'true' }}
+    needs: installation
+    runs-on: self-hosted
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+            
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: go.mod
+            
+      - name: Placeholder
+        run: |
+          echo "Placeholder for import tests"
+
+  delete-cluster:
+    if: ${{ always() && github.event.inputs.keep_cluster != 'true' }}
+    needs: [provisioning-tests, importing-tests, support-matrix-provisioning-tests, support-matrix-importing-tests]
+    runs-on: self-hosted
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+            
+      - name: Delete k3s cluster
+        run: |
+          make clean-all
 
   delete-runner:
-    if: ${{ always() && needs.create-runner.result == 'success' && inputs.destroy_runner == true }}
-    needs:
-      - create-runner
-      - e2e-tests
+    if: ${{ always() && inputs.destroy_runner == true }}
+    needs: delete-cluster
```

Comment:
Looking at this condition, I wonder if we need `delete-cluster` at all. If we want to keep the cluster running, we will also keep the runner up. In which case, simply having `delete-runner` should suffice. WDYT?

Reply:
Correct, fixed it.